Problems/1929/ans_1929.py

```diff
@@ -0,0 +1,21 @@
+#https://www.acmicpc.net/problem/1929
+
+#에라토스테네스의 체를 통해 소수 구하기
+
+def prime_number(N):
+    # prime_max=1
+    prime_list = [False]*2+[True]*(N-1)
+    # while prime_max**2<=N:
+        # prime_max+=1  -> int(N**0.5)로 대체
+    for i in range(2,int(N**0.5)+1):
+        if prime_list[i]:
+            # for j in range(2*i,N+1,i):
+                # prime_list[j] = False -> slicing으로 대체
+            prime_list[2*i::i] = [False]*(N//i-1)
```

Comment:
for문 처리가 이해하기 어려웠음.
```suggestion
for i, v in enumerate(prime_list):
    if i*i > N:
        break
    if v:
        prime_list[2*i::i] = [False]*((N+1//i)-1) 
```
List slicing은 좋은 기능이긴 하다만, 이런식으로 set 하는 방식으로 사용하면
갯수가 맞지 않을 경우에 대해서 고려를 해야 함.
알고리즘 상, N//i -1 을 했어도 유효 input 범위 안에서 어쨋든 false가 나오기 때문에 괜찮았지만,
len(prime_list) == N+1 이므로 알고리즘에 따라서는 error 발생 할 수 있어보임.
list slicing은 reference로만 사용하는게 좋은 것 같다는게 내 결론인데,
다른 사람들은 어떻게 쓰는지 잘 모르겠네.

Reply:
enumerate 함수 이용한 갯수 고려하는 방법 확인! 감사~